src/formatDate.js

```diff
@@ -8,7 +8,37 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separateDate = date.split(fromFormat[3]);
+  const newDate = [];
+  const separator = toFormat[3];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    let index = toFormat.indexOf(fromFormat[i]);
+
+    if (index === -1) {
+      if (fromFormat[i].length === 2) {
+        index = toFormat.indexOf('YYYY');
+
+        if (separateDate[i] >= 30) {
+          separateDate[i] = '19' + separateDate[i];
+        }
+
+        if (separateDate[i] < 30) {
+          separateDate[i] = '20' + separateDate[i];
+        }
+      }
+
+      if (fromFormat[i].length === 4) {
+        index = toFormat.indexOf('YY');
+
+        separateDate[i] = separateDate[i].slice(2);
+      }
+    }
+
+    newDate[index] = separateDate[i];
```

Comment:
There is a potential issue here if 'index' is -1, which can happen if the 'toFormat' array does not contain the expected format. This could lead to 'newDate' having undefined values. Consider adding a check to ensure 'index' is valid before assigning 'separateDate[i]' to 'newDate[index]'.